### PR TITLE
New example - warpPerspective_demo.cpp

### PR DESCRIPTION
**Pull request by sturkmen72**
### This pullrequest changes

An example program shows using cv::findHomography and cv::warpPerspective for image warping

based on a sample code http://study.marearts.com/2015/03/image-warping-using-opencv.html

### CODE REVIEW

File: samples/cpp/warpPerspective_demo.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    if (event == EVENT_LBUTTONUP)
+    {
+        dragging = false;
+    }
+
+    if (event == EVENT_RBUTTONUP)
+    {
+        roi4point[0].x = 365;
```
Comment (sovrasov) [inline samples/cpp/warpPerspective_demo.cpp]: These hard-coded values look strange and for some images are not suitable. It's better to use relative quantities, depending on image width/height.

File: samples/cpp/warpPerspective_demo.cpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        //prepare to get homography matrix
+        vector< Point2f> P1; //clicked positions
+        vector< Point2f> P2(4); //user setting positions
+        for (int i = 0; i < 4; ++i)
+            P1.push_back(roi4point[i]);
+
+        //user setting position
+        P2[0].x = 0;
+        P2[0].y = 0;
+        P2[1].x = 300;
```
Comment (sovrasov) [inline samples/cpp/warpPerspective_demo.cpp]: The same thing about hard-coded values.

**Comment by sturkmen72** [comment]
@sovrasov i am still working on some improvements on the code. your comments was in my todo list
thanks for reviewing.

**Comment by sovrasov** [comment]
:+1:

**Pull request merged**